packages/packages/core/editor-components/src/component-id-transformer.ts

```diff
@@ -4,15 +4,23 @@ import { __getState as getState } from '@elementor/store';
 import { selectUnpublishedComponents } from './store/store';
 import { getComponentDocumentData } from './utils/component-document-data';
 
-export const componentIdTransformer = createTransformer( async ( id: number ) => {
-	const unpublishedComponents = selectUnpublishedComponents( getState() );
+type PublishedComponentId = number;
+type UnpublishedComponentId = string;
+
+type ComponentId = PublishedComponentId | UnpublishedComponentId;
 
-	const unpublishedComponent = unpublishedComponents.find( ( component ) => component.id === id );
+export const componentIdTransformer = createTransformer( async ( id: ComponentId ) => {
+	const unpublishedComponents = selectUnpublishedComponents( getState() );
 
+	const unpublishedComponent = unpublishedComponents.find( ( component ) => component.uid === id );
```

Comment:
While at it, maybe this would be nicer -
```suggestion
	const unpublishedComponent = unpublishedComponents.find( ( { uid } ) => uid === id );
```
?